src/commands/key2ds.rs

```diff
@@ -0,0 +1,166 @@
+use std::{fs::File, io::Write, path::PathBuf};
+
+use clap::{builder::ValueParser, Parser};
+use domain::{
+    base::{
+        iana::{DigestAlg, SecAlg},
+        zonefile_fmt::ZonefileFmt,
+        Record,
+    },
+    rdata::Ds,
+    validate::DnskeyExt,
+    zonefile::inplace::{Entry, ScannedRecordData},
+};
+
+use crate::error::Error;
+
+#[derive(Clone, Debug, Parser)]
+#[command(version)]
+pub struct Key2ds {
+    /// ignore SEP flag (i.e. make DS records for any key)
+    #[arg(short = 'f')]
+    ignore_sep: bool,
+
+    /// do not write DS records to file(s) but to stdout
+    #[arg(short = 'n')]
+    write_to_stdout: bool,
+
+    /// use SHA1 for the DS hash
+    #[arg(short = '1', overrides_with_all = ["one", "two", "four", "algorithm"])]
+    one: bool,
+
+    /// use SHA256 for the DS hash
+    #[arg(short = '2', overrides_with_all = ["one", "two", "four", "algorithm"])]
+    two: bool,
+
+    /// use SHA384 for the DS hash
```

Comment:
Maybe change the algorithm names to be the mnemonics, so that the algorithms in the help output and the arguments for `--algorithm` are the same: `SHA-1`, `SHA-256`, and `SHA-384`. I got confused for a moment when trying to test the algorithms ^^

Reply:
These arguments are only there for compatibility. So we're out of luck here :)

We could (later) refactor this with separate argument parsing (like I did for notify).

Reply:
Yes, the arguments need to stay for compatibility. I was only confused by the help output saying the algorithm is spelled `SHA1`, and then tried using it with `--algorithm` while testing. The mnemonics for the hash algorithms for DS digests are with dashes (e.g. SHA-1) (while the DNSKEY algorithms don't use delimiters (e.g. RSASHA1)).
I'm just trying to avoid confusion for people trying to migrate to using `--algorithm` or `-a`.

```suggestion
    /// use SHA-1 for the DS hash
    #[arg(short = '1', overrides_with_all = ["one", "two", "four", "algorithm"])]
    one: bool,

    /// use SHA-256 for the DS hash
    #[arg(short = '2', overrides_with_all = ["one", "two", "four", "algorithm"])]
    two: bool,

    /// use SHA-384 for the DS hash
```

Reply:
Ah gotcha! Will fix